modules/library/main/src/main/java/org/geotools/api/data/DataAccessFactory.java

```diff
@@ -584,4 +584,14 @@ public String toString() {
             return buf.toString();
         }
     }
+
+    @SuppressWarnings("unchecked")
+    static <T> T lookup(Param param, Map<String, ?> params, Class<T> target) throws IOException {
```

Comment:
Please add Javadocs, since this method became public after begin extracted from a subclass.